molgenis-data-cache/src/main/java/org/molgenis/data/cache/L1/L1CacheRepositoryDecorator.java

```diff
@@ -0,0 +1,306 @@
+	}
+
+	@Override
+	public void deleteById(Object id)
+	{
+		if (cacheable) l1Cache.evict(getName(), id);
+		decoratedRepository.deleteById(id);
+	}
+
+	@Override
+	public void deleteAll()
+	{
+		if (cacheable) l1Cache.evictAll(getName());
+		decoratedRepository.deleteAll();
+	}
+
+	@Override
+	public void deleteAll(Stream<Object> ids)
+	{
+		if (cacheable)
+		{
+			String entityName = getName();
+			ids = ids.filter(id -> {
+				l1Cache.evict(entityName, id);
+				return true;
+			});
+		}
+		decoratedRepository.deleteAll(ids);
+	}
+
+	@Override
+	public void close() throws IOException
+	{
+		decoratedRepository.close();
+	}
+
+	@Override
+	public void forEachBatched(Fetch fetch, Consumer<List<Entity>> consumer, int batchSize)
+	{
+		decoratedRepository.forEachBatched(fetch, consumer, batchSize);
+	}
+
+	@Override
+	public Set<RepositoryCapability> getCapabilities()
+	{
+		return decoratedRepository.getCapabilities();
+	}
+
+	@Override
+	public Set<QueryRule.Operator> getQueryOperators()
+	{
+		return decoratedRepository.getQueryOperators();
+	}
+
+	@Override
+	public String getName()
+	{
+		return decoratedRepository.getName();
+	}
+
+	@Override
+	public EntityMetaData getEntityMetaData()
+	{
+		return decoratedRepository.getEntityMetaData();
+	}
+
+	@Override
+	public Iterator<Entity> iterator()
+	{
+		return decoratedRepository.iterator();
+	}
+
+	@Override
+	public long count()
+	{
+		return decoratedRepository.count();
+	}
+
+	@Override
+	public Query query()
+	{
+		return decoratedRepository.query();
+	}
+
+	@Override
+	public long count(Query q)
+	{
+		return decoratedRepository.count(q);
+	}
+
+	@Override
+	public Stream<Entity> findAll(Query q)
+	{
+		return decoratedRepository.findAll(q);
+	}
+
+	@Override
+	public Entity findOne(Query q)
+	{
+		return decoratedRepository.findOne(q);
+	}
+
+	@Override
+	public Stream<Entity> findAll(Stream<Object> ids, Fetch fetch)
+	{
+		return decoratedRepository.findAll(ids, fetch);
+	}
+
+	@Override
+	public AggregateResult aggregate(AggregateQuery aggregateQuery)
+	{
+		return decoratedRepository.aggregate(aggregateQuery);
+	}
+
+	@Override
+	public void flush()
+	{
+		decoratedRepository.flush();
+	}
+
+	@Override
+	public void clearCache()
+	{
+		decoratedRepository.clearCache();
+	}
+
+	@Override
+	public void rebuildIndex()
+	{
+		decoratedRepository.rebuildIndex();
+	}
+
+	@Override
+	public void addEntityListener(EntityListener entityListener)
+	{
+		decoratedRepository.addEntityListener(entityListener);
+	}
+
+	@Override
+	public void removeEntityListener(EntityListener entityListener)
+	{
+		decoratedRepository.removeEntityListener(entityListener);
+	}
+
+	/**
+	 * Looks up the Entities for a List of entity IDs.
+	 * Those present in the cache are returned from cache. The missing ones are retrieved from the decoratedRepository.
+	 *
+	 * @param batch list of entity IDs to look up
+	 * @return List of {@link Entity}s
+	 */
+	private List<Entity> findAllBatch(List<Object> batch)
```

Comment:
@Mark-de-Haan @fdlk Can we use Guava CacheLoader instead of writing code ourselves?